### ISSUE TITLE
Feature Normalization Layers

### ISSUE THREAD

**Issue by wenkesj** [fomorians/pyoneer#8]
# Feature Request
Normalization layers that TF does not directly support as a layer such as:
+ [GroupNormalization](https://arxiv.org/abs/1803.08494)
+ [LayerNormalization](https://arxiv.org/abs/1607.06450)

## Objective
Make it easier, Eager-friendly, and intuitive to define statistical normalization operations _other than_ `BatchNormalization` as a `tf.layers.Layer`.

## Motivation
TF does not make it easy to use other normalization operations as a `tf.layers.Layer`. There are existing implementations using the `variable_scope` methodology for `graph`-mode layers, but not class oriented layers that are easier to use with Eager.

### Use Cases
ResNet with GroupNormalization will improve performance on small batch sizes. `LayerNormalization` improves RNN/LSTM performance as a nice BatchNormalization performance replacement.

## Design Proposal
Follow the same API as the TF [BatchNormalization](https://www.tensorflow.org/api_docs/python/tf/layers/BatchNormalization) implementation using the existing [group_norm](https://www.tensorflow.org/api_docs/python/tf/contrib/layers/group_norm) and [layer_norm](https://www.tensorflow.org/api_docs/python/tf/contrib/layers/layer_norm) implementations.

### Examples
```python
class LayerNormFC(tf.keras.Model):

  def __init__(self):
    self.h0 = tf.layers.Dense(64)
    self.ln0 = pynr.layers.LayerNormalization()

  def call(self, inputs, training=False):
    outputs = self.h0(inputs)
    outputs = self.ln0(outputs, training=training)
    outputs = tf.nn.relu(outputs)
    return outputs
```

### API
Follow [BatchNormalization](https://www.tensorflow.org/api_docs/python/tf/layers/BatchNormalization).

### Implementation Plan
Follow [BatchNormalization](https://www.tensorflow.org/api_docs/python/tf/layers/BatchNormalization) for `LayerNormalization` and `GroupNormalization` as part of the `pynr.layers` module.

### User Migration
N/A

### Relation to Related Features
#7 and #3 talk about this very briefly and indirectly.

### Alternatives Considered
Use the native TF contrib API or BatchNormalization axes tricks.

**Comment by jimfleming** [comment]
Closing. These are in TF[1] and TF Addons[2].

1. https://www.tensorflow.org/versions/r2.0/api_docs/python/tf/keras/layers/LayerNormalization
2. https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/normalizations.py
